pkg/server/service/internalhandler.go

```diff
@@ -44,37 +42,35 @@ func NewInternalHandlers(api func(configuration *runtime.Configuration) http.Han
 	}
 }
 
-// internalWithModifier wraps an internal handler together with a response modifier.
-// Its goal is to apply the modifier on the response that would be served by the internal handler,
-// but before the response is actually written to the original response writer.
-// It is safe to call ServeHTTP on an internalWithModifier with a nil modifier.
-type internalWithModifier struct {
-	internal http.Handler
+type internalResponseWriterWithModifier struct {
```

Comment:
it could now be used with any kind of rw, so no need to specify "internal" anymore in the name. I would suggest simply: responseModifier. Also I think there should be at least a go doc for this part:

// It can be safely used with a nil modifier.

Reply:
I applied the changes you requested